syntax = "proto2";

package asn1_types;

message Identifier {
  required Class id_class = 2;
  required Encoding encoding = 1;
}

// See X.690 (2015), 8.1.2.2.
enum Class {Universal = 0;
  Application = 1;
  ContextSpecific = 2;
  Private = 3;
}

// See X.690 (2015), 8.1.2.5.
enum Encoding {
  Primitive = 0;
  Constructed = 1;
}

message ASN1BitString {
  required Identifier id = 1;
  required string val = 2;
}

message ASN1PrintableString {
  required Identifier id = 1;
  required string val = 2;
}

message ASN1IA5String {
  required Identifier id = 1;
  required string val = 2;
}

message ASN1Integer {
  required Identifier id = 1;
  required bytes val = 2;
}

message ASN1Boolean {
  required Identifier id = 1;
  required bool val = 2;
}